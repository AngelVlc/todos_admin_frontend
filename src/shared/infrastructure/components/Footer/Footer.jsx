import React from "react";

export const Footer = () => {
  return (
    <footer className="footer py-1">
      <div className="content has-text-right">
        <span className="is-size-7">{`sha: ${process.env.REACT_APP_COMMIT_SHA}, build: ${process.env.REACT_APP_BUILD_DATE}`}</span>
      </div>
    </footer>
  );
};
