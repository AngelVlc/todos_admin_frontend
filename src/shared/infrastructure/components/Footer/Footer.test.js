import { render, cleanup } from '@testing-library/react';
import { Footer } from './Footer';

const renderComponent = () => {
  return {
    ...render(
      <Footer/>
    ),
  };
};

afterEach(cleanup);

const originalEnv = process.env;

beforeEach(() => {
  jest.resetModules()
  process.env = { ...originalEnv };
});

afterAll(() => {
  process.env = originalEnv;
});

it("Should match the snapshot", () => {
  process.env.REACT_APP_COMMIT_SHA = 'sha'
  process.env.REACT_APP_BUILD_DATE = 'date'

  const { asFragment } = renderComponent();

  expect(asFragment(<Footer />)).toMatchSnapshot();
});
