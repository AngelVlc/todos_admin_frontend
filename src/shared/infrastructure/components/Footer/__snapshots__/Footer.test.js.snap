// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer py-1"
  >
    <div
      class="content has-text-right"
    >
      <span
        class="is-size-7"
      >
        sha: sha, build: date
      </span>
    </div>
  </footer>
</DocumentFragment>
`;
